### ISSUE TITLE
Nested Values Disappear when using StrictMode

### ISSUE THREAD

**Issue by MAST1999** [TanStack/form#685]
### Describe the bug

I have a problem that if I use it with react-aria's TextField they don't work (the TextField receives undefined.)

Based on The [discord discussion](https://discord.com/channels/719702312431386674/1100437019857014895/1232729283370287246) and my local testing, it seems to be a StrictMode problem.

### Your minimal, reproducible example

https://stackblitz.com/edit/vitejs-vite-hwpxmr?file=package.json,src%2FApp.tsx,src%2Findex.css&terminal=dev

### Steps to reproduce

Open the reproduction link and see the inputs are empty.

### Expected behavior

The inputs should be showing values (they have values set in the form.)

### How often does this bug happen?

Every time

### Platform

OS: Linux
Browser: Chrome
Version: 124.0.63

### TanStack Form adapter

react-form

### TanStack Form version

^0.19.2

### TypeScript version

^5.4.5

**Comment by crutchcorn** [comment]
With some testing, this only appears to happen with nested values: `name="one.two"` type deal.

Investigating now, this is a huge p1 issue

**Comment by crutchcorn** [comment]
OK, I think I narrowed it down to a single LOC issue. Making a PR momentarily, once I verify that the rest of our tests pass

**Comment by crutchcorn** [comment]
Fixed in #686 with tests to verify. Thanks for reporting and filing!

**Comment by dlindahl** [comment]
I am also seeing an issue with `StrictMode`, specifically with using `mode=array` with a list of strings. In `StrictMode`, adding a new value with `pushValue` appears to not do anything. Under the hood, I believe it is adding the value and then immediately reverting the field value to whatever the previous value was.

I took the array example from the docs and changed it in the following ways:

1. Wrap `App` in `StrictMode`.
2. Change `[Array<{ name: string; age: number }>](people: [] as Array<{ name: string; age: number }>)` to `people: [] as Array<String>`

Clicking on the "Add person" button does not result in a new row.

Also of note is when you leave the `people` value to be an array of objects, `StrictMode` works fine. Its only when the underlying value is an array of _strings_ that it breaks. If I had to guess, I'd say there is a value check somewhere that is failing since an empty string is "falsey", but I have no idea what the root cause would be.

Reproduction case: https://stackblitz.com/edit/tanstack-form-rxy9kq?file=src%2Findex.tsx&preset=node
